src/main/java/bio/terra/app/controller/ApiValidationExceptionHandler.java

```diff
@@ -66,6 +69,25 @@
         return new ResponseEntity<>(errorModel, HttpStatus.BAD_REQUEST);
     }
 
+    @Override
+    protected ResponseEntity<Object> handleTypeMismatch(
+        TypeMismatchException ex, HttpHeaders headers, HttpStatus status, WebRequest request) {
+        String message = status + " - see error details";
+        List<String> details = Collections.emptyList();
+        for (Throwable cause = ex.getCause(); cause != null; cause = cause.getCause()) {
+            if (cause instanceof DataRepoException) {
+                message = cause.getMessage();
+                details = ((DataRepoException) cause).getErrorDetails();
```

Comment:
`break` after this line? Otherwise you keep looping and could overwrite this information.

Reply:
That's actually the intended effect. I'm looking to get the root cause and only return that info.

Reply:
OK, so can you loop until `cause.getCause() == null` and only convert the data in that `cause`?